app/src/main/scala/org/alephium/explorer/persistence/queries/TransactionQueries.scala

```diff
@@ -409,9 +396,22 @@ trait TransactionQueries extends StrictLogging {
     getBalanceQueryDEPRECATED(address).map(sumBalance)
   }
 
-  def getBalanceAction(address: Address): DBActionR[(U256, U256)] = {
-    getBalanceQuery(address).map(sumBalance)
-  }
+  def getBalanceAction(address: Address): DBActionR[(U256, U256)] =
+    sql"""
+      SELECT sum(outputs.amount),
+             sum(CASE
+                     WHEN outputs.lock_time is NULL or outputs.lock_time < (extract(epoch from now()) * 1000) THEN 0
```

Comment:
Will `epoch from now()` be called multiple times in this query? If so, we could make `now` as a query parameter. Pseudo code:
```
def getBalanceAction(address: Address, now: TimeStamp): DBActionR[(U256, U256)] =
    sql"""
      SELECT sum(outputs.amount),
             sum(CASE
                     WHEN outputs.lock_time is NULL or outputs.lock_time < $now THEN 0
```

Reply:
Yes it will be called once. To confirm here is a test. 

```sql
-- Create over a million rows (1,078,591) in the output table
select count(*) from outputs; 

-- This query returns a single row with time value 1651467106.505805
select distinct (time) from (select extract(epoch from now()) as time from outputs) inner_query; 
```
Still gonna go with your suggestion here 👍🏼  because with time passed as parameter to the query we don't need to do timezone conversions.

Reply:
Nice catch of the conversion issue 👍